src/colmap/controllers/incremental_mapper.cc

```diff
@@ -248,6 +262,78 @@ bool IncrementalMapperController::LoadDatabase() {
     return false;
   }
 
+  // If prior positions are to be used and setup from the database, convert
+  // geographic coords. to cartesian ones
+  if (options_->use_prior_position &&
+      options_->set_prior_position_from_database) {
+    return SetupPriorPoseFromDatabase();
+  }
+
+  return true;
+}
+
+bool IncrementalMapperController::SetupPriorPoseFromDatabase() {
+  LOG(INFO) << "Setting up prior positions";
+
+  Timer timer;
+  timer.Start();
+  std::size_t nb_prior_positions = 0;
+  bool prior_is_gps = true;
+  std::vector<Eigen::Vector3d> v_gps_prior;
+  image_t ref_image_id = std::numeric_limits<image_t>::max();
+  double ref_lat = std::numeric_limits<double>::max();
+  double ref_lon = std::numeric_limits<double>::max();
+  v_gps_prior.reserve(database_cache_->NumImages());
+  for (const auto& image : database_cache_->Images()) {
+    if (image.second.HasPosePrior()) {
+      ++nb_prior_positions;
+      if (image.second.WorldFromCamPrior().coordinate_system !=
+          PosePrior::CoordinateSystem::WGS84) {
+        prior_is_gps = false;
+        // break;
+      } else {
+        v_gps_prior.push_back(image.second.WorldFromCamPrior().position);
+        // Image with the lowest id is to be used as the origin for prior
```

Comment:
To minimize the approximation error of ENU, you might want to use the average position instead of an arbitrary one.

Reply:
I agree that using the average position could be slightly more accurate.  However, using the prior of the image with the lowest id in the database as the reference allows users to easily know which GPS measurement has been used as reference.  This might be valuable to users that need to convert back the estimated position into GPS (lat/lon) data.

Reply:
Moreover, the model_aligner function already uses the prior of the image with the lowest id in the database as the reference.  So doing the same here allows to have coherent results between both features.

Reply:
This makes sense, thank you.